### ISSUE TITLE
Feature request: Clear_collected should remove episodes from Trakt Collection that were collected in the first place

### ISSUE THREAD

**Issue by NGDM** [Taxel/PlexTraktSync#1452]
### Feature description
The clear_collected feature should clear the items that were present in your collection but aren't anymore.
Now it is seemingly bulk-clearing all episodes of a show that aren't present in the Plex library. 
This generates a lot of unnecessary API calls as well as messages in the log.

### Use case

This would make the script more efficient/performant and make health-checking and debugging using the logs easier. 

### Current behaviour in logs

```text
2023-04-06 10:49:57,137 INFO[PlexTraktSync]:Remove from Trakt collection (1/3190): <TVEpisode>: Malcolm in the Middle S0E1 A Stroke of Genius
2023-04-06 10:49:57,183 INFO[PlexTraktSync]:Remove from Trakt collection (2/3190): <TVEpisode>: Malcolm in the Middle S0E2 Outtakes
2023-04-06 10:49:57,189 INFO[PlexTraktSync]:Remove from Trakt collection (3/3190): <TVEpisode>: Malcolm in the Middle S0E3 Gag Reel
2023-04-06 10:49:57,193 INFO[PlexTraktSync]:Remove from Trakt collection (4/3190): <TVEpisode>: Malcolm in the Middle S0E4 Malcolm Vision
2023-04-06 10:49:57,196 INFO[PlexTraktSync]:Remove from Trakt collection (5/3190): <TVEpisode>: Malcolm in the Middle S0E5 DS Deals And Sellouts
2023-04-06 10:49:57,199 INFO[PlexTraktSync]:Remove from Trakt collection (6/3190): <TVEpisode>: Malcolm in the Middle S0E6 Cold Opens
2023-04-06 10:49:57,202 INFO[PlexTraktSync]:Remove from Trakt collection (7/3190): <TVEpisode>: Malcolm in the Middle S0E7 Season One Promos
2023-04-06 10:49:57,205 INFO[PlexTraktSync]:Remove from Trakt collection (8/3190): <TVEpisode>: Malcolm in the Middle S0E8 Malcolm: Behind the Middle
2023-04-06 10:49:57,209 INFO[PlexTraktSync]:Remove from Trakt collection (9/3190): <TVEpisode>: Malcolm in the Middle S0E9 Dewey's Day Job: A Portrait of Erik Per Sullivan
2023-04-06 10:49:57,213 INFO[PlexTraktSync]:Remove from Trakt collection (10/3190): <TVEpisode>: Malcolm in the Middle S0E10 Pilot Extended
2023-04-06 10:49:57,240 INFO[PlexTraktSync]:Remove from Trakt collection (11/3190): <TVEpisode>: Malcolm in the Middle S0E11 None
2023-04-06 10:49:57,243 INFO[PlexTraktSync]:Remove from Trakt collection (12/3190): <TVEpisode>: Malcolm in the Middle S0E12 Company Picnic 1: Director's cut (Season 3, episode 11)
2023-04-06 10:49:57,247 INFO[PlexTraktSync]:Remove from Trakt collection (13/3190): <TVEpisode>: Malcolm in the Middle S0E13 Quarantunes Reunion
2023-04-06 10:49:57,251 INFO[PlexTraktSync]:Remove from Trakt collection (14/3190): <TVEpisode>: The Mandalorian S3E7 Chapter 23
2023-04-06 10:49:57,256 INFO[PlexTraktSync]:Remove from Trakt collection (15/3190): <TVEpisode>: The Mandalorian S3E8 Chapter 24
2023-04-06 10:49:57,260 INFO[PlexTraktSync]:Remove from Trakt collection (16/3190): <TVEpisode>: Ted Lasso S0E1 The Missing Christmas Mustache
2023-04-06 10:49:57,263 INFO[PlexTraktSync]:Remove from Trakt collection (17/3190): <TVEpisode>: Ted Lasso S3E5 Signs
2023-04-06 10:49:57,266 INFO[PlexTraktSync]:Remove from Trakt collection (18/3190): <TVEpisode>: Ted Lasso S3E6 Every Disadvantage Has Its Advantage
2023-04-06 10:49:57,269 INFO[PlexTraktSync]:Remove from Trakt collection (19/3190): <TVEpisode>: Ted Lasso S3E7 Ola's
2023-04-06 10:49:57,272 INFO[PlexTraktSync]:Remove from Trakt collection (20/3190): <TVEpisode>: Ted Lasso S3E8 We'll Never Have Paris
2023-04-06 10:49:57,275 INFO[PlexTraktSync]:Remove from Trakt collection (21/3190): <TVEpisode>: Ted Lasso S3E9 The Omission Attrition
2023-04-06 10:49:57,278 INFO[PlexTraktSync]:Remove from Trakt collection (22/3190): <TVEpisode>: Ted Lasso S3E10 Episode 10
2023-04-06 10:49:57,282 INFO[PlexTraktSync]:Remove from Trakt collection (23/3190): <TVEpisode>: Ted Lasso S3E11 Episode 11
2023-04-06 10:49:57,284 INFO[PlexTraktSync]:Remove from Trakt collection (24/3190): <TVEpisode>: Ted Lasso S3E12 Episode 12
```

### Plan to implement
No, little python experience

**Comment by glensc** [comment]
You can enable debug logging, then request urls get logged to the log.

From my logs, I think this is the request for shows
- https://api-v2launch.trakt.tv/users/me/collection/shows?extended=metadata

you can probably make requests with trakt apiary:
- https://trakt.docs.apiary.io/

I myself use this script:
```sh
$ cat curl.sh
#!/bin/sh

curl \
     --header "Content-Type: application/json" \
     --header "trakt-api-version: 2" \
     --header "trakt-api-key: $CLIENT_ID" \
     --header "Authorization: Bearer $OAUTH_TOKEN" \
	"$@"
```

```sh
./curl.sh "https://api-v2launch.trakt.tv/users/me/collection/shows?extended=metadata" | jq -C | less
```

**Comment by twolaw** [comment]
Because of this bug : https://github.com/moogar0880/PyTrakt/issues/76

Explained briefly :

`TVSeason` init() expects `season` as the season number variable.

But the unpacking of `/users/me/collection/shows?extended=metadata` returns the season number in a variable named `number`.
So the `TVSeason` will always init season 1 (default) because `season` value is unknown.
Then `TVShow` will only have the season 1.

**Comment by glensc** [comment]
Fixed with 0.26.2
